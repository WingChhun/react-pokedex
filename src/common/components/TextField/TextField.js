import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import _noop from 'lodash/noop';


const Input = styled.input`
  border:2px solid red;
  width:100%;
  `;


function TextField(props) {
  const [value, setValue] = useState(props.value);

  const handleChange = (e) => setValue(e.target.value);

  return (
    <Input value={value} onChange={handleChange} />
  )
};


TextField.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string
};

TextField.defaultProps = {
  value: '',
  onChange: _noop()
};


export default TextField;