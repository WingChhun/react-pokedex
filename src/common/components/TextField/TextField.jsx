import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import _noop from 'lodash/noop';
import { PokemonContext } from '../../context';

const Input = styled.input`
  border: 2px solid red;
  width: 100%;
  padding: 8px 12px;
`;

export const PokemonTextField = props => {
  const [{ filterStr }, dispatch] = useContext(PokemonContext);

  return (
    <TextField
      value={filterStr}
      onChange={value => dispatch({ type: 'CHANGE_FILTER', payload: value })}
    />
  );
};

export const TextField = props => {
  const { value, type, onChange, ...rest } = props;

  return (
    <Input
      type={type}
      value={value}
      onChange={e => onChange(e.target.value)}
      {...rest}
    />
  );
};

TextField.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string,
  type: PropTypes.string,
  context: PropTypes.string
};

TextField.defaultProps = {
  value: '',
  type: 'text',
  onChange: _noop()
};
