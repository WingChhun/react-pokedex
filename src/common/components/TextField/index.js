export { TextField, PokemonTextField } from './TextField';
