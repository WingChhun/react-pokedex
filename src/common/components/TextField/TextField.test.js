import React from 'react';
import TextField from "./TextField";
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() })


//todo: test hooks
describe(`<TextField/>`, () => {

  it(`renders properly`, () => {
    const wrapper = shallow(<TextField />);
    expect(wrapper.toMatchSnapshot());
  })
});