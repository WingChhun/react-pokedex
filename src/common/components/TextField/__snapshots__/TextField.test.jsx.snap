// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField/> renders properly 1`] = `ShallowWrapper {}`;
